src/plugins/ERC20TokenLimitPlugin.sol

```diff
@@ -0,0 +1,170 @@
+
+import {PackedUserOperation} from "@eth-infinitism/account-abstraction/interfaces/PackedUserOperation.sol";
+import {UserOperationLib} from "@eth-infinitism/account-abstraction/core/UserOperationLib.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {IAccountExecute} from "@eth-infinitism/account-abstraction/interfaces/IAccountExecute.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {PluginManifest, PluginMetadata} from "../interfaces/IPlugin.sol";
+import {IStandardExecutor, Call} from "../interfaces/IStandardExecutor.sol";
+import {IPlugin} from "../interfaces/IPlugin.sol";
+import {IExecutionHook} from "../interfaces/IExecutionHook.sol";
+import {BasePlugin, IERC165} from "./BasePlugin.sol";
+
+/// @title ERC20 Token Limit Plugin
+/// @author ERC-6900 Authors
+/// @notice This plugin supports an ERC20 token spend limit. This should be combined with a contract whitelist
+/// plugin to make sure that token transfers not tracked by the plugin don't happen.
+/// Note: this plugin is opinionated on what selectors can be called for token contracts to guard against weird
+/// edge cases like DAI. You wouldn't be able to use uni v2 pairs directly as the pair contract is also the LP
+/// token contract
+contract ERC20TokenLimitPlugin is BasePlugin, IExecutionHook {
+    using UserOperationLib for PackedUserOperation;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    string public constant NAME = "ERC20 Token Limit Plugin";
+    string public constant VERSION = "1.0.0";
+    string public constant AUTHOR = "ERC-6900 Authors";
+
+    struct ERC20SpendLimit {
+        address token;
+        uint256[] limits;
+    }
+
+    mapping(address account => mapping(address => uint256[] limits)) public limits;
+    mapping(address account => EnumerableSet.AddressSet) internal _tokenList;
+
+    error ExceededNativeTokenLimit();
+    error ExceededNumberOfEntities();
+    error SelectorNotAllowed();
+
+    function getTokensForAccount(address account) external view returns (address[] memory tokens) {
+        tokens = new address[](_tokenList[account].length());
+        for (uint256 i = 0; i < _tokenList[account].length(); i++) {
+            tokens[i] = _tokenList[account].at(i);
+        }
+        return tokens;
+    }
+
+    function updateLimits(uint8 functionId, address token, uint256 newLimit) external {
+        _tokenList[msg.sender].add(token);
+        limits[msg.sender][token][functionId] = newLimit;
+    }
+
+    function _checkAndDecrementLimit(uint8 functionId, bytes4 selector, uint256 spend, address token) internal {
+        if (selector == IERC20.transfer.selector || selector == IERC20.approve.selector) {
+            uint256 limit = limits[msg.sender][token][functionId];
+            if (spend > limit) {
+                revert ExceededNativeTokenLimit();
+            }
+            limits[msg.sender][token][functionId] = limit - spend;
```

Comment:
@adamegyed This linter warning is curious - I couldn't find an external function invoked before _checkAndDecrementLimit. Do you see any?

Reply:
my guess is that it looks for `transfer` word, or selector instead of the function call.. we have that in L57. Don't think theres any way around it, have to just silence this warning